write-fonts/src/tables/variations.rs

```diff
@@ -39,3 +270,77 @@ impl<'a> FromObjRef<Option<read_fonts::tables::variations::Tuple<'a>>> for Vec<F
             .unwrap_or_default()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn point_pack_words() {
+        let thing = PackedPointNumbers {
+            is_all: false,
+            numbers: vec![1002, 2002, 8408, 12228],
+        };
+
+        let runs = thing.iter_runs().collect::<Vec<_>>();
+        assert!(runs[0].are_words);
+        assert_eq!(runs[0].last_point, 0);
+        assert_eq!(runs[0].points, &[1002, 2002, 8408, 12228]);
+
+        let bytes = crate::dump_table(&thing).unwrap();
+        let (read, _) = read_fonts::tables::variations::PackedPointNumbers::split_off_front(
+            FontData::new(&bytes),
+        );
+        assert_eq!(thing.numbers, read.iter().collect::<Vec<_>>());
+    }
+
+    #[test]
+    fn smoke_test_point_packing() {
+        let thing = PackedPointNumbers {
+            is_all: false,
+            numbers: vec![5, 25, 225, 1002, 2002, 2008, 2228],
+        };
+
+        let runs = thing.iter_runs().collect::<Vec<_>>();
+        assert!(!runs[0].are_words);
+        assert_eq!(runs[0].last_point, 0);
+        assert_eq!(runs[0].points, &[5, 25, 225]);
+
+        assert!(runs[1].are_words);
+        assert_eq!(runs[1].last_point, 225);
+        assert_eq!(runs[1].points, &[1002, 2002]);
+
+        assert!(!runs[2].are_words);
+        assert_eq!(runs[2].last_point, 2002);
+        assert_eq!(runs[2].points, &[2008, 2228]);
+
+        assert_eq!(runs.len(), 3);
```

Comment:
here too I'd rather you assert the full contents of runs for the nicer diff and to have the expected value grouped up instead of mixed with the accessors to get thigns to compare to.

Reply:
I prefer this style because when a failure occurs, you know immediately what field of which run you need to look at, instead of trying to visually read a big diff to find the one incorrect field.